### ISSUE TITLE
Bad handling of audio metadata in scrobbler (+crashes)

### ISSUE THREAD

**Issue by szmer** [zaps166/QMPlay2#222]
Thank you for making this: this is my preferred Linux media player for some time interface and featurewise, but it comes with some annoyances. I am on Fedora 28/29, GNOME 3, with QMP2 18.12.16. I haven't seen in recent changelog anything related to these problems:

1. The program interprets non-empty "Album Artist" in audio metadata as "Album Title" when scrobbling to Last.fm. (My current workaround is blanking album artist in EasyTAG, whenever I remember to do this).
2. The scrobbler can't handle special characters. For example, if there is "&" somewhere in album name, it won't scrobble it properly -- nothing will show up in my Last.fm profile. It will also flash "Logged in to Last.fm" in status bar again and again. And it crashes after playing each track with this issue.

It would be also nice if there was a way to explicitly save your library/playlist without restarting the program. As of now, I restart every time I add something to my collection, because the changes are gone if the whole program crashes.

**Comment by zaps166** [comment]
I'll check lastfm - the code is very old and not tested for a long time.
I can add a key shortcut for saving playlist. Or maybe it is better to do a backup file?

**Comment by szmer** [comment]
> I can add a key shortcut for saving playlist. Or maybe it is better to do a backup file?

...you're right, I checked and you can save it to PLS file, and then reload and it does preserve the catalog structure. I haven't made the mental connection that I can do this with my big "whole library" playlist :P And there is even a shortcut. So it's a question whether you want an idiot-friendly button for saving the default playlist specifically.

**Comment by zaps166** [comment]
> The scrobbler can't handle special characters (...)

This should be fixed now, could you test it?

> The program interprets non-empty "Album Artist" in audio metadata as "Album Title" when scrobbling to Last.fm.

Could you upload an example file?

**Comment by zaps166** [comment]
> Could you upload an example file?

I've reproduced it using EasyTAG.

**Comment by zaps166** [comment]
> The program interprets non-empty "Album Artist" in audio metadata as "Album Title" (...)

Should be also fixed

**Comment by zaps166** [comment]
Please reopen this if something is still wrong with LastFM.

If playlist needs to be saved somehow before correct application quit please open a new issue (now QMPlay2 shouldn't crash using LastFM).

**Comment by szmer** [comment]
I just managed to verify that everything is working fine now, thanks!

It took me quite some time over last days to build, going through some problems with compilation (just to semi-document if someone encounters similar):
- had to install Qt 5 from their website and export `CMAKE_PREFIX_PATH=$HOME/Qt/5.12.2/gcc_64` (ie. the correct catalog in the installation folder), otherwise CMake complained about lack of QtWidgets configuration
- then there was a linking error on compilation that looked like this:
```
...
[ 88%] Built target Notify
[ 88%] Linking CXX executable QMPlay2
//usr/lib64/libavutil.so.56: undefined reference to `clReleaseMemObject@OPENCL_1.0'
//usr/lib64/libavutil.so.56: undefined reference to `clReleaseCommandQueue@OPENCL_1.0'
//usr/lib64/libavutil.so.56: undefined reference to `clCreateContext@OPENCL_1.0'
... (etc.)
```
I suspect that it has something to do with proprietary NVIDIA drivers, which I'm forced to use. At least this suggests so: https://github.com/OpenKinect/libfreenect2/issues/804 - I did what they said and also set `LD_LIBRARY_PATH=/usr/lib64/:$LD_LIBRARY_PATH` (I found this path with `locate libOpenCL`).
- And then, to run `QMPlay2` from terminal, I had to remove `/usr/lib64/` from `LD_LIBRARY_PATH` and *append* to it `$HOME/Qt/5.12.2/gcc_64/lib`.
